sfcasts/ep1/github-service-part-1.md

```diff
@@ -1,62 +1,69 @@
-status of healthy or sick. Now let's go back to our test. Oops, let's go back to our
-test and we need to add the use statement for our GitHub service. So the best way to
-do this is chop off the last couple of letters and there we go. GitHub service app
-service. That's the one we want. And PHP storm adds the use statement for us.
-
-We can go back to our terminal and let's run the test one more time. Make sure that
-yes, instead of two failures, we only have one. Our GitHub service test test GitHub
-test get health report returns, correct health status for Dino with data set sick
-Dino, it failed to asserting that two variables reference the same object. And it's
-of course doing that because we are only returning a healthy status for get health
-reports. Now that we have this kind of wired up, let's go ahead and implement the
-service.
+# GitHub Service - Part 1
 
+Now that we can see if a `Dinosaur` is accepting visitors on our dashboard, we
+need to keep the dashboard updated in real time by using the health status labels
+that GenLab has applied to several dino issues on GitHub. To do that we'll create 
+a service that will grab those labels from the issues using GitHub's API.
+
+## Test for our service first
+
+To test our new service, inside of `tests/Unit/` create a new `Service/` directory 
+and then a new class: `GithubServiceTest` which will extend `TestCase`. Add method
+`testGetHealthReportReturnsCorrectHealthStatusForDino` and inside, even
+though it doesn't exist yet, `$service = new GithubService()`.
+
+Our service will return a `HealthStatus` enum that is created from the health status
+label on GitHub, so we'll `assertSame()` that `$expectedStatus` is identical to 
+`$service->getHealthReport()`. Yes, we'll be using a data provider for this test...
+And because our service will return a *single* `HealthStatus` object, let's tell 
+the method which dino we need a health report for by passing in a `$dinoName` to
+`getHealthReport()`.
+
+Create a new method: `dinoNameProvider()` that returns a `\Generator`. Our first 
+dataset for the provider will have the key `Sick Dino`, which returns an array 
+with `HealthStatus::SICK` and `Daisy` for the dino's name. Next up is a 
+`Healthy Dino` with `HealthStatus::HEALTHY` who happens to be the one and only
+`Maverick`. On the test method, add a `@dataProvider` annotation so the test uses 
+the `dinoNameProvider` and we'll also pass in a `HealthStatus $expectedStatus`
+argument and a `string` called `$dinoName`.
+
+Move to your terminal and let's check out our tests:
+
+```terminal
+./vendor/bin/phpunit
+```
+
+And... Yup! Just as I expected, we have two errors because: 
+
+> The GithubService class cannot be found
+
+## Create the service that will call GitHub
+
+To fix those errors, inside of `src/`, create a new `Service/` directory. Then 
+we'll need a new class: `GithubService` and inside, add a method: `getHealthReport()`
```

Comment:
```diff
a new class
the new class
```

We aren't creating a class that could be called anything, we know EXACTLY which class we need to create. We are creating THE class we need.